ecommerce/views.py

```diff
@@ -99,6 +99,7 @@ def get_queryset(self):
                     & Q(content_type__model="courserun")
                 )
             )
+            .order_by("programs__title", "course_run__course__title")
```

Comment:
@briangrossman does this order make sense to you -- sorting first by program title and then by course title?

Reply:
Hmmm... my gut is that this isn't quite right...

... the non-technical answer to this question is that the sorting should be exactly the order that results in an alphabetized list in the pulldown. Whatever is being displayed in the pulldown (sometimes it displays programs, sometimes it displays courses) should be alphbetized.

Reply:
@pdpinch your point makes sense but Since this view returns a mixed list of products containing a content object of either a program or a course run at for each list entry due to generic relation of the product with Program and CourseRun. On the xPro frontend, this list is used as two sublists for depending upon the `product_type`, this is where the order of program/course titles matters in the sublists. Though we have the option to go with conditional annotate here as well to populate the results. Will also wait for @ahmed-arbisoft for the review and see what changes he might suggest as well.

Reply:
@pdpinch @briangrossman I think this should work. We only have two types of entities here: a) `CourseRun` and b) `Program`, for sorting. We do not show any courses in a program, nor do we show which program a course run belongs to. From a development perspective it looks wrong, sorting on program title first and then course title but in practice we show these entities separately (by selecting "Course" or "Program" in the dropdown) that gets rid of any other intermediate entities of the other kind (included due to simply ordering alphabetically). Visually the response would be like:

1. A first program
2. B first course run
3. C second course run
4. D second program

But when it's filtered on the front-end by entity type we end up with either:
1. B first course run
2. C second course run

or:
1. A first program
2. D second program

Which for our use case should suffice.